### PR TITLE
Feature/e UI 4362 display list of bookings

### PR DESCRIPTION
**Pull request by tarikdogan-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-4362

### Change description ###

display list of bookings current an feature ones

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by Mo-Lala84 — APPROVED** [review]
approved

File: src/booking/containers/booking-home/booking-home.component.spec.ts [hunk 1]
```diff
@@ -64,6 +85,38 @@ describe('BookingHomeComponent', () => {
     expect(bookingsList).toBeTruthy();
   });
 
+  it('should display bookings in sorted when there are existing bookings and first radio button is clicked', () => {
```
Comment (paulhowes-HMCTS) [inline src/booking/containers/booking-home/booking-home.component.spec.ts:88]: in sorted when = in sorted order

File: src/booking/containers/booking-home/booking-home.component.ts [hunk 2]
```diff
@@ -30,33 +31,54 @@ export class BookingHomeComponent implements OnInit, OnDestroy {
...
-    this.existingBookingsSubscription = this.bookingService.getBookings().subscribe(response => {
-      this.existingBookings = response.bookings;
-      this.assignBookingLocation();
+    this.existingBookingsSubscription = this.bookingService.getBookings().subscribe((arr) => {
+      this.locations$ = arr.bookings.map(
+        booking =>  this.bookingService.getBookingLocation(booking.base_location_id).pipe(
+          map( location => {
+            return {
+              ...booking,
+              locationName: location[0] && !!location[0].building_location_name && location[0].building_location_name  } ;
```
Comment (paulhowes-HMCTS) [inline src/booking/containers/booking-home/booking-home.component.ts:40]: Why you have the double bang check as well as the location name?

File: src/booking/containers/booking-home/booking-home.component.ts [hunk 3]
```diff
@@ -30,33 +31,54 @@ export class BookingHomeComponent implements OnInit, OnDestroy {
...
 
-  public assignBookingLocation() {
-    this.existingBookings.forEach(booking => {
-      this.bookingLocationSubscription = this.bookingService.getBookingLocation(booking.base_location_id).subscribe(location =>
-        booking.locationName = location[0] && location[0].building_location_name ?
-                                    location[0].building_location_name :
-                                    null
-      );
-    });
+  private orderByCurrentThenFeature() {
```
Comment (paulhowes-HMCTS) [inline src/booking/containers/booking-home/booking-home.component.ts:57]: Think you mean orderByCurrentThenFuture (all occurrences)

**Review by paulhowes-HMCTS — CHANGES_REQUESTED** [review]
A couple of minor tweaks and a query pertaining to the !!.

File: src/booking/containers/booking-home/booking-home.component.ts (hunk 2 above)
Comment (tarikdogan-hmcts) [reply src/booking/containers/booking-home/booking-home.component.ts:40]: checking for undefined, if it's not undefined that return from service then assign location name

Comment (paulhowes-HMCTS) [reply src/booking/containers/booking-home/booking-home.component.ts:40]: Does that mean there instances where it would be undefined (as opposed to null)?

Comment (tarikdogan-hmcts) [reply src/booking/containers/booking-home/booking-home.component.ts:40]: no Now we are using mock service but actual service implementation may return null or undefined. So good to check undefined as a standard way as in >=ES5  .

File: src/booking/containers/booking-home/booking-home.component.spec.ts (hunk 1 above)
Comment (paulhowes-HMCTS) [reply src/booking/containers/booking-home/booking-home.component.spec.ts]: Suggestion now applied so resolving the conversation

File: src/booking/containers/booking-home/booking-home.component.ts (hunk 2 above)
Comment (paulhowes-HMCTS) [reply src/booking/containers/booking-home/booking-home.component.ts]: Double bang removed as per discussion so resolving conversation

File: src/booking/containers/booking-home/booking-home.component.ts (hunk 3 above)
Comment (paulhowes-HMCTS) [reply src/booking/containers/booking-home/booking-home.component.ts]: Changed to future so resolving conversation

**Review by paulhowes-HMCTS — APPROVED** [review]
Approving based on feedback and amends

**Pull request merged**